x/oracle/keeper/operator.go

```diff
@@ -83,29 +83,39 @@ func (k Keeper) CreateOperator(ctx sdk.Context, address sdk.AccAddress, collater
 }
 
 // RemoveOperator removes an operator, creates an withdrawal for collateral and gives back rewards immediately.
-func (k Keeper) RemoveOperator(ctx sdk.Context, address sdk.AccAddress) error {
-	if !k.IsOperator(ctx, address) {
+func (k Keeper) RemoveOperator(ctx sdk.Context, operatorAddress, proposerAddress string) error {
+	// Ensure that the sender of the tx is either the operator to be removed itself or a certifier.
+	proposerAddr, err := sdk.AccAddressFromBech32(proposerAddress)
+	if err != nil {
+		return err
+	}
+	if operatorAddress != proposerAddress || !k.CertKeeper.IsCertifier(ctx, proposerAddr) {
```

Comment:
needs to be `and`?

Reply:
Thank you

Reply:
Do we need the same checks for operator creation as well, or just for removal?

Reply:
good point. I'm not sure why we have the field `Operator.Proposer`. Should we get rid of it?

Reply:
I think it was intended for proposing and approving it but I think we can just make creation free of any constraints